app/controllers/passwordless/sessions_controller.rb

```diff
@@ -42,28 +42,40 @@ def create
     def show
       # Make it "slow" on purpose to make brute-force attacks more of a hassle
       BCrypt::Password.create(params[:token])
-
-      destination =
-        Passwordless.redirect_back_after_sign_in &&
-        reset_passwordless_redirect_location!(authenticatable_class)
-
       sign_in passwordless_session
 
-      redirect_to destination || main_app.root_path
+      redirect_to passwordless_success_redirect_path
     rescue Errors::TokenAlreadyClaimedError
       flash[:error] = I18n.t(".passwordless.sessions.create.token_claimed")
-      redirect_to main_app.root_path
+      redirect_to passwordless_failure_redirect_path
     rescue Errors::SessionTimedOutError
       flash[:error] = I18n.t(".passwordless.sessions.create.session_expired")
-      redirect_to main_app.root_path
+      redirect_to passwordless_failure_redirect_path
     end
 
     # match '/sign_out', via: %i[get delete].
     #   Signs user out. Redirects to root_path
     # @see ControllerHelpers#sign_out
     def destroy
       sign_out authenticatable_class
-      redirect_to main_app.root_path
+      redirect_to passwordless_sign_out_redirect_path
+    end
+
+    protected
+
+    def passwordless_sign_out_redirect_path
+      Passwordless.sign_out_redirect_path
+    end
+
+    def passwordless_failure_redirect_path
+      Passwordless.failure_redirect_path
+    end
+
+    def passwordless_success_redirect_path
+      return Passwordless.success_redirect_path unless Passwordless.redirect_back_after_sign_in
+
+      redirect_url = reset_passwordless_redirect_location!(authenticatable_class)
+      params[:url] || redirect_url || Passwordless.success_redirect_path
```

Comment:
A fair point, I've added some code which strips out the host and schema from any URI that is passed in, so that it will just be used as a path redirect.